connaisseur/flask_application.py

```diff
@@ -116,14 +98,62 @@ def readyz():
     return "", 200
 
 
+async def __async_mutate():
+    # Maximum timeout for admission control is 30s
+    # If Connaisseur issues requests that timeout after the webhook times out
+    # k8s will report that Connaisseur was unresponsive, which isn't quite true
+    # thus we timeout slightly earlier
+    # https://github.com/sse-secure-systems/connaisseur/issues/448
+    timeout = aiohttp.ClientTimeout(total=const.AIO_TIMEOUT_SECONDS)
+    async with aiohttp.ClientSession(timeout=timeout) as session:
+        admission_request = None
+        try:
+            logging.debug(request.json)
+            admission_request = AdmissionRequest(request.json)
+            response = await __admit(admission_request, session)
+            # Depending on whether alerting must succeed, run it synchronously or in the background
+            if alerting_must_succeed(True):
```

Comment:
I fail to see why you call with `True`

Reply:
`event_category = "admit_request" if admit_event else "reject_request"` in `alerting_must_succeed`. Could make it more specific and call it `alerting_must_succeed_if_admitted`, but didn't really see the benefit here of making it more specifc. Thoughts?

Reply:
Though I shall rename the function to must_... to reflect the bool answer

Reply:
Indeed, it is never called with False. Writing it to answer to any bool was easier (copy pasta from other parts), so I didn't feel the need to make the function less generic.

However, thinking about it, you're right. It really doesn't make sense to be called with False ever as the question is whether Conny should reject the image if alert sending failed and if Conny rejected in the first place, this is irrelevant.

Another question that came up while looking at this is whether we'd want to dispatch the alerts in the "rejected" case in the background as well. Because even in the rejected case we may run into a timeout (though it's less critical in the sense that we'd block the deployment anyways whereas a blocked deployment due to alerts being sent too slow would change behaviour). So I think I'll fix that up as well and send alerts in the background here too

Reply:
sounds good